### PR TITLE
Improved performance of string token value unescape logic by handling…

### PR DESCRIPTION
**Pull request by erictraut**
… the common cases (no format string and no escape characters) using a fast path.

### CODE REVIEW

**Comment by erictraut** [comment]
@jakebailey, let me know if you'd like me to merge this into the master branch or hold it on a side branch until after the next pylance release.

File: packages/pyright-internal/src/parser/stringTokenUtils.ts [hunk 1]
```diff
@@ -83,8 +83,40 @@ function completeUnescapedString(incomplete: IncompleteUnescapedString): Unescap
...
+    if (isRaw) {
+        return {
+            value: escapedString,
+            unescapeErrors: [],
+            nonAsciiInBytes: false,
+            formatStringSegments: [],
+        };
+    }
+
+    const charCodes: number[] = [];
```
Comment (bschnurr) [inline packages/pyright-internal/src/parser/stringTokenUtils.ts:96]: maybe there is a minor benefit to pre-allocate the array size. 

something like 
```
const charCodes: number[] = new Array(escapedString.length);
```

Comment (jakebailey) [reply packages/pyright-internal/src/parser/stringTokenUtils.ts:96]: That won't do quite what you expect:

```
> const arr = new Array(5)
undefined
> arr.push("test")
6
> arr
[ <5 empty items>, 'test' ]
>
```

Array is weird and has "empty items" as a thing. Maybe preallocs are possible by doing the above and setting `arr.length = 0`, but that would require profiling to confirm.

**Review by jakebailey — APPROVED** [review]
LGTM, and seems safe to me.

**Pull request merged**